app/gui/src/controller/searcher.rs

```diff
@@ -2209,64 +1711,57 @@ pub mod test {
 
         // Typing more args by hand.
-        searcher.set_input("testFunction1 some_arg pat".to_string()).unwrap();
-        let (func,) = frags_borrow().iter().cloned().expect_tuple();
-        assert_eq!(func.id, CompletedFragmentId::Function);
-        assert!(are_same(&func.picked_suggestion, &entry1));
+        searcher.set_input("testFunction1 some_arg pat".to_string(), Byte(26)).unwrap();
+        let (func,) = picked_suggestions().iter().cloned().expect_tuple();
+        assert!(are_same(&func.entry, &test_function_1));
 
         // Picking argument's suggestion.
-        let new_input =
-            searcher.use_suggestion(action::Suggestion::FromDatabase(entry2.clone_ref())).unwrap();
-        assert_eq!(new_input, "testFunction1 some_arg test_var_1 ");
-        let new_input =
-            searcher.use_suggestion(action::Suggestion::FromDatabase(entry2.clone_ref())).unwrap();
-        assert_eq!(new_input, "testFunction1 some_arg test_var_1 test_var_1 ");
-        let (function, arg1, arg2) = frags_borrow().iter().cloned().expect_tuple();
-        assert_eq!(function.id, CompletedFragmentId::Function);
-        assert!(are_same(&function.picked_suggestion, &entry1));
-        assert_eq!(arg1.id, CompletedFragmentId::Argument { index: 1 });
-        assert!(are_same(&arg1.picked_suggestion, &entry2));
-        assert_eq!(arg2.id, CompletedFragmentId::Argument { index: 2 });
-        assert!(are_same(&arg2.picked_suggestion, &entry2));
-
-        // Backspacing back to the second arg.
-        searcher.set_input("testFunction1 some_arg test_var_1 test_v".to_string()).unwrap();
-        let (picked, arg) = frags_borrow().iter().cloned().expect_tuple();
-        assert_eq!(picked.id, CompletedFragmentId::Function);
-        assert!(are_same(&picked.picked_suggestion, &entry1));
-        assert_eq!(arg.id, CompletedFragmentId::Argument { index: 1 });
-        assert!(are_same(&arg.picked_suggestion, &entry2));
-
-        // Editing the picked function.
-        searcher.set_input("testFunction2 some_arg test_var_1 test_v".to_string()).unwrap();
-        let (arg,) = frags_borrow().iter().cloned().expect_tuple();
-        assert_eq!(arg.id, CompletedFragmentId::Argument { index: 1 });
-        assert!(are_same(&arg.picked_suggestion, &entry2));
-    }
-
-    #[wasm_bindgen_test]
+        let suggestion1 = action::Suggestion::FromDatabase(test_var_1.clone_ref());
+        let new_input = searcher.use_suggestion(suggestion1.clone()).unwrap();
+        assert_eq!(new_input.text, "test_var_1 ");
+        let new_input = searcher.use_suggestion(suggestion1).unwrap();
+        assert_eq!(new_input.text, "test_var_1 ");
+        let (function, arg1, arg2) = picked_suggestions().iter().cloned().expect_tuple();
+        assert!(are_same(&function.entry, &test_function_1));
+        assert!(are_same(&arg1.entry, &test_var_1));
+        assert!(are_same(&arg2.entry, &test_var_1));
+
+        // TODO
```

Comment:
I don't understand what's going on in this part of the test, I will fix it or remove it.

Reply:
Seems it is not compatible with the new behavior, removed this piece